.github/workflows/validate-k8s-manifests.yml

```diff
@@ -0,0 +1,233 @@
+
+  render_head_ref_charts:
+    runs-on: ubuntu-latest
+    needs: get_changed_helm_charts
+    strategy:
+      matrix:
+        chart: ${{ fromJSON(needs.get_changed_helm_charts.outputs.matrix_charts) }}
+    steps:
+      - name: checkout repository
+        uses: actions/checkout@v4
+
+      - name: setup helm
+        uses: azure/setup-helm@v4.0.0
+
+      - name: render ${{ matrix.chart }} from head ref
+        id: render_head
+        run: |
+          mkdir -p shared/head-charts
+          git fetch origin "$HEAD_REF"
+          git checkout  "$HEAD_REF" --
+          if [ -f "${{ matrix.chart }}/Chart.yaml" ]; then
+            helm dependency update "${{ matrix.chart }}"
+            values_files="${{ matrix.chart }}"/values-*
+            for values_file in $(basename -a $values_files); do
+              env_name=$(basename "$values_file" | sed -E 's/^values-(.+)\.ya?ml$/\1/')
+              helm template "${{ matrix.chart }}" -f "${{ matrix.chart }}/values.yaml" -f "${{ matrix.chart }}/${values_file}" --output-dir "shared/charts/${{ matrix.chart }}/${env_name}"
+            done
+          fi
+          echo sanitized_name=$(echo "${{ matrix.chart }}" | sed 's/\//-/g') >> $GITHUB_OUTPUT
+      - name: upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: "shared-head-${{ steps.render_head.outputs.sanitized_name }}"
+          path: "shared"
+
+  validate_rendered_helm_chart_manifests:
+    runs-on: ubuntu-latest
+    needs:
+      - get_changed_helm_charts
+      - render_head_ref_charts
+    steps:
+      - name: install kubeconform
+        run: |
+          set -euo pipefail
+
+          FILENAME="kubeconform-linux-amd64.tar.gz"
+          DOWNLOAD_URL="${KUBECONFORM_BASE_URL}/v${KUBECONFORM_VERSION}/${FILENAME}"
+
+          echo "Downloading kubeconform from ${DOWNLOAD_URL}..."
+          wget -q "$DOWNLOAD_URL" -O "$FILENAME"
+
+          echo "${KUBECONFORM_SHA256}  $FILENAME" | sha256sum -c -
+
+          echo "Extracting kubeconform..."
+          tar -xzf "$FILENAME"
+          chmod +x kubeconform
+          sudo mv kubeconform /usr/local/bin/kubeconform
+
+      - name: download artifacts
+        uses: actions/download-artifact@v4
+        with:
+          pattern: shared-*
+          merge-multiple: true
+          path: "shared"
+
+      - name: validate k8s manifests
+        id: validate_k8s_manifests
+        run: |
+          set -eo pipefail
+
+          charts="${{ needs.get_changed_helm_charts.outputs.charts }}"
+          exit_code=0
+
+          for chart in $charts; do
+            echo "Validating ${chart}..."
+            if [ -d "shared/charts/${chart}" ]; then
+              if ! /usr/local/bin/kubeconform \
+                  -schema-location default \
+                  -schema-location "${KUBECONFORM_SCHEMA_LOCATION}" \
+                  -summary "shared/charts/${chart}" >> kubeconform.log 2>&1; then
```

Comment:
I have no idea if this could help you down below with the processing but it looks like you can set the output to json? Then maybe you could use JQ to parse and clean it upt?

Reply:
https://github.com/yannh/kubeconform?tab=readme-ov-file#usage